scilab-bin: adding application launchers

fusion809 opened the pull request:
Moving the desktop configuration files from `opt/scilab-$ver/share/applications` to `share/applications`, making the required changes, and moving the icons in `opt/scilab-$ver/share/icons` to `share/icons`.

###### Motivation for this change
Presently the scilab-bin package provides no application launchers for the Scilab, Scinotes and Xcos apps. In this PR I am adding these launchers. 

On lines 76, 79 and 82 I am replacing `Terminal=false` with `Terminal=true` in the Scilab, Scinotes and Xcos desktop config files, as I've tried launching these apps using these launchers, without the `Terminal=true` change, and they always crash on start up. Changing to `Terminal=true` fixes this problem for all three apps. 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

worldofpeace commented on pkgs/applications/science/math/scilab-bin/default.nix:
```diff
@@ -65,6 +65,31 @@ stdenv.mkDerivation rec {
     ln -s "scilab-${ver}" "$out/bin/scilab-${majorVer}"
+
+    mkdir -p "$out/share/applications"
+    mv "$out/opt/scilab-${ver}/share/applications/scilab-adv-cli.desktop" "$out/share/applications"
```
Can we not do `mv $out/opt/scilab-${ver}/share/applications/*.desktop $out/share/applications`?

fusion809 replied:
Not a bad idea, that's exactly what I started out with, but it (the error from mv) said no file was found.

worldofpeace replied:
That could have been because of the quoting.

fusion809 replied:
Ah, you were right, thanks!

worldofpeace reviewed (commented):
Can you separate your commands, with comments, into groups like:

* moving desktop files
* moving icons
* correcting content of desktop files

Without that visual hierarchy (or something similar) it's kind of thick to read.

Also why not using `substituteInPlace` instead of sed directly?

fusion809 commented:
Largely used sed as I know how to use it, I don't know how to use substituteInPlace. I'll add the comments.

worldofpeace commented:
> Largely used sed as I know how to use it, I don't know how to use substituteInPlace.

substituteInPlace usually looks like:

```bash
substituteInPlace file \
  --replace pattern replacement
```

Though it isn't a problem if you don't use it.

fusion809 commented:
Are those changes satisfactory?

worldofpeace commented:
> Are those changes satisfactory?

I think so. I'm verifying locally now.

fusion809 commented:
My last commit is essentially me making things more concise.

worldofpeace commented:
$out/opt/scilab-6.0.1/bin/scilab-cli doesn't execute and by extension the desktop file doesn't launch the application.

fusion809 commented:
Aha, lol, forgot to create the needed symlink.

fusion809 commented:
While I'm at it, do you know why there's `scilab-${ver}` and `scilab-${majorVer}` symlinks created in `$out/bin`? What purpose do these two symlinks serve?

worldofpeace commented:
> While I'm at it, do you know why there's `scilab-${ver}` and `scilab-${majorVer}` symlinks created in `$out/bin`? What purpose do these two symlinks serve?

I don't really know. You could try removing them and see if it works. Git blame reveals that it's always been here.

fusion809 commented:
Ah, that is unexpected.

worldofpeace commented:
Would you mind also moving `$out/opt/scilab-6.0.1/share/{appdata,locale,mime}` to `$out/share?` Though I'm not certain about `locale`.

worldofpeace commented on pkgs/applications/science/math/scilab-bin/default.nix:
```diff
@@ -61,10 +61,40 @@ stdenv.mkDerivation rec {
+    rm -rf $out/opt/scilab-${ver}/share/{applications,icons}
+
+    # Moving other share/ folders
+    mv $out/opt/scilab-6.0.1/share/{appdata,locale,mime} $out/share
```
```suggestion
    mv $out/opt/scilab-${ver}/share/{appdata,locale,mime} $out/share
```

fusion809 replied:
Oops, sorry.

fusion809 committed:
scilab-bin: adding application launchers
Moving the desktop configuration files from
opt/scilab-$ver/share/applications to share/applications, making the
required changes, and moving the icons in opt/scilab-$ver/share/icons to
share/icons. Moving locales, appdata and mime out of there
opt/scilab-${ver} too. Also removing unneeded scilab-${ver} and
scilab-${majorVer} symlinks from bin/.

worldofpeace commented:
Now I'm going to check if it builds FTR

@GrahamcOfBorg build scilab-bin

worldofpeace commented:
@fusion809 Thanks for contributing :sparkles:
